galsim/inclinedsersic.py

```diff
@@ -0,0 +1,230 @@
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""@file inclinedsersic.py
+
+InclinedSersic is a class representing a (possibly-truncated) Sersic profile inclined to the LOS.
+"""
+
+from galsim import GSObject
+import galsim
+
+from . import _galsim
+
+
+class InclinedSersic(GSObject):
+    """A class describing an inclined sersic profile. This class is general, and so for certain
+    special cases, more specialized classes will be more efficient. For the case where n==1
+    with no truncation, the InclinedExponential class will be much more efficient. For the case
+    where the inclination angle is zero (face-on), the Sersic class will be slightly more efficient.
+
+    The Inclined Sersic surface brightness profile is characterized by four properties: its
+    Sersic index `n', its inclination angle (where 0 degrees = face-on and 90 degrees = edge-on),
+    its scale radius, and its scale height. The 3D light distribution function is:
+
+        I(R,z) = I_0 / (2h_s) * sech^2 (z/h_s) * exp[-b*(R/r_s)^{1/n}]
+
+    where z is the distance along the minor axis, R is the radial distance from the minor axis,
+    r_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk. The 2D light distribution function is then determined
+    from the scale height and radius here, along with the inclination angle.
+
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+
+    At present, this profile is not enabled for photon-shooting.
+
+    The allowed range of values for the `n` parameter is 0.3 <= n <= 6.2.  An exception will be
+    thrown if you provide a value outside that range, matching the range of the Sersic profile.
+
+    This class shares the caching of Hankel transformations with the Sersic class; see that
+    class for documentation on efficiency considerations with regards to caching.
+
+    A profile can be initialized using one (and only one) of two possible size parameters:
+    `scale_radius` or `half_light_radius`.  Exactly one of these two is required. Similarly,
+    at most one of `scale_height' and `scale_h_over_r' is required; if neither is given, the
```

Comment:
Should clarify both here and below (in the parameter list) that even if the user chooses to specify the half-light radius, ```scale_h_over_r``` still uses scale radius. Unless I've misunderstood that?